course_UvA-DL/inception-resnet-densenet/Inception_ResNet_DenseNet.py

```diff
@@ -0,0 +1,1298 @@
+                    "max": 16
+                },
+                act_fn=self.hparams.act_fn
+            ),
+            InceptionBlock(
+                128,
+                c_red={
+                    "3x3": 48,
+                    "5x5": 16
+                },
+                c_out={
+                    "1x1": 32,
+                    "3x3": 64,
+                    "5x5": 16,
+                    "max": 16
+                },
+                act_fn=self.hparams.act_fn
+            )
+        )
+        # Mapping to classification output
+        self.output_net = nn.Sequential(
+            nn.AdaptiveAvgPool2d((1, 1)), nn.Flatten(), nn.Linear(128, self.hparams.num_classes)
+        )
+
+    def _init_params(self):
+        # Based on our discussion in Tutorial 4, we should initialize the
+        # convolutions according to the activation function
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                nn.init.kaiming_normal_(m.weight, nonlinearity=self.hparams.act_fn_name)
+            elif isinstance(m, nn.BatchNorm2d):
+                nn.init.constant_(m.weight, 1)
+                nn.init.constant_(m.bias, 0)
+
+    def forward(self, x):
+        x = self.input_net(x)
+        x = self.inception_blocks(x)
+        x = self.output_net(x)
+        return x
+
+
+# %% [markdown]
+# Now, we can integrate our model to the model dictionary we defined above:
+
+# %%
+model_dict["GoogleNet"] = GoogleNet
+
+# %% [markdown]
+# The training of the model is handled by PyTorch Lightning, and we just have to define the command to start.
+# Note that we train for almost 200 epochs, which takes about an hour on Lisa's default GPUs (GTX1080Ti).
+# We would recommend using the saved models and train your own model if you are interested.
+
+# %%
+googlenet_model, googlenet_results = train_model(
+    model_name="GoogleNet",
+    model_hparams={
+        "num_classes": 10,
+        "act_fn_name": "relu"
+    },
+    optimizer_name="Adam",
+    optimizer_hparams={
+        "lr": 1e-3,
+        "weight_decay": 1e-4
+    }
+)
+
+# %% [markdown]
+# We will compare the results later in the notebooks, but we can already print them here for a first glance:
+
+# %%
+print("GoogleNet Results", googlenet_results)
+
+# %% [markdown]
+# ### Tensorboard log
+#
+# A nice extra of PyTorch Lightning is the automatic logging into TensorBoard.
+# To give you a better intuition of what TensorBoard can be used, we can look at the board that PyTorch Lightning has been generated when training the GoogleNet.
+# TensorBoard provides an inline functionality for Jupyter notebooks, and we use it here:
+
+# %%
+# Import tensorboard
+# %load_ext tensorboard
+
+# %%
+# Opens tensorboard in notebook. Adjust the path to your CHECKPOINT_PATH!
+# %tensorboard --logdir ../saved_models/tutorial5/tensorboards/GoogleNet/
+
+# %% [markdown]
+# <center width="100%"><img src="tensorboard_screenshot_GoogleNet.png" width="1000px"></center>
```

Comment:
as MD?

Reply:
Yes, I will try to write a small script that replaces the HTML expressions for png images with MD syntax.